src/file_storage.cpp

```diff
@@ -1218,7 +1218,6 @@ namespace {
 				new_mtime.push_back(0);
 
 			auto& file = new_files.back();
-			TORRENT_ASSERT(static_cast<std::int64_t>(file.size) <= max_file_size);
```

Comment:
this assert didn't make sense, since `file.size` is already a 48 bit bitfield, this check is always true